src/webnn/native/dml/GraphDML.cpp

```diff
@@ -316,15 +316,18 @@ namespace webnn::native::dml {
                     input = ::dml::Reinterpret(input, newInputDims,
                                                CalculateInputLayoutStrides(NhwcToNchw, inputDims));
                     break;
-                case NchwToNhwc:
+                case NchwToNhwc: {
                     newInputDims.resize(4);
                     newInputDims[0] = inputDims[0];
                     newInputDims[1] = inputDims[2];
                     newInputDims[2] = inputDims[3];
                     newInputDims[3] = inputDims[1];
-                    input = ::dml::Reinterpret(input, newInputDims,
-                                               CalculateInputLayoutStrides(NchwToNhwc, inputDims));
-                    break;
+                    ::dml::Optional<::dml::TensorStrides> strides = input.GetOutputDesc().strides;
+                    DAWN_ASSERT(strides != ::dml::NullOpt);
+                    ::dml::TensorStrides newStrides = {strides.value()[0], strides.value()[2],
```

Comment:
Please keep the same code stype for `newInputDims`
`::dml::TensorDimensions newInputDims = {};`

Reply:
i mean you can assign it with the code `::dml::TensorDimensions newInputDims = {inputDims[0], inputDims[2], inputDims[3], inputDims[1]};`

Reply:
OK, I will. Thanks! @fujunwei